runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/Work.java

```diff
@@ -111,15 +120,58 @@ public Collection<Windmill.LatencyAttribution> getLatencyAttributions() {
       if (duration.equals(Duration.ZERO)) {
         continue;
       }
-      list.add(
-          Windmill.LatencyAttribution.newBuilder()
-              .setState(state)
-              .setTotalDurationMillis(duration.getMillis())
-              .build());
+      LatencyAttribution.Builder laBuilder = Windmill.LatencyAttribution.newBuilder();
+      if (state == LatencyAttribution.State.ACTIVE) {
+        laBuilder = addActiveLatencyBreakdownToBuilder(isHeartbeat, laBuilder, workId, sampler);
+      }
+      Windmill.LatencyAttribution la =
+          laBuilder.setState(state).setTotalDurationMillis(duration.getMillis()).build();
+      list.add(la);
     }
     return list;
   }
 
+  private static LatencyAttribution.Builder addActiveLatencyBreakdownToBuilder(
+      Boolean isHeartbeat,
```

Comment:
unless isHeartbeat is nullable, prefer to use the primitive type (boolean) to the boxed. go/java-practices/primitives#prefer-unboxed

Reply:
1636e774170b95f004943fa7e80175044f06b206